### PR TITLE
Fix quantity update of related tickets

### LINKED ISSUE #2938: Quantity does not update for newly assigned Tickets when editing the capacity of the related date in multi-step
Steps:
1. Edit a date which has related tickets
2. In edit form, change capacity to lower than the quantity of any ticket that's not related to this date
3. Click **Save and assign tickets**
4. Assign the ticket mentioned in step 2 to this date
5. Submit

Expected:
Quantity of the newly assigned ticket should be set to the capacity of the edited date.

Actual:
Quantity of the newly assigned ticket doesn't change at all

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Fixes the quantity update of newly assigned tickets in date multi-step form
- Fixes the quantity update of newly assigned tickets in TAM

Closes #2938

### CODE REVIEW

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts [hunk 1]
```diff
@@ -42,16 +57,30 @@ const useOnSubmitAssignments = () => {
 			if (ticketsToUpdate.length < datesToUpdate.length) {
 				ticketsToUpdate.forEach(([id, possibleRelation]) => {
 					const datetimes = possibleRelation?.datetimes || [];
-					updateTicket({ id, datetimes });
+					const input: UpdateTicketInput = { id, datetimes };
+					// if an entry exists in changed quantity map
+					// lets use this oppurtunity to update the quantity here
+					// to reduce the number of mutation requests
+					if (ticketsWithChangedQuantity?.[id]) {
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts:64]: let's give this a name?

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts [hunk 2]
```diff
@@ -42,16 +57,30 @@ const useOnSubmitAssignments = () => {
...
+						input.quantity = ticketsWithChangedQuantity?.[id];
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts:65]: this is used twice, so let's give it a name, so we don't have to check twice which is what

Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts:65]: Done

Comment (tn3rb) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts]: @nicolad 

like this?

```suggestion
					const newTicketQuantity = ticketsWithChangedQuantity?.[id]
					if (newTicketQuantity) {
						input.quantity = newTicketQuantity;
```

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts [hunk 3]
```diff
@@ -42,16 +57,30 @@ const useOnSubmitAssignments = () => {
...
+						// mark the ticket as already updated
+						updatedTickets.push(id);
```
Comment (tn3rb) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts:69]: where is this checked?

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts [hunk 4]
```diff
@@ -42,16 +57,30 @@ const useOnSubmitAssignments = () => {
...
 				datesToUpdate.forEach(([id, possibleRelation]) => {
 					const tickets = possibleRelation?.tickets || [];
 					updateDatetime({ id, tickets });
 				});
 			}
+
+			// now we finally update the ticket quantities
+			Object.entries(ticketsWithChangedQuantity).forEach(([id, quantity]) => {
+				updateTicket({ id, quantity });
+			});
```
Comment (tn3rb) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts]: do we need to check `updatedTickets` first ?

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils.ts [hunk 5]
```diff
@@ -36,6 +37,76 @@ export const prepareEntitiesForUpdate = <Entity extends TAMRelationEntity>({
...
+		// if capacity is infinite or it's more than ticket quantity
+		if (isInfinite(minimumCapacity) || minimumCapacity > nonNegativeTicketQuantity) {
+			// no need to update the ticket quantity
+			return acc;
+		}
+
+		// set the quantity to minimum capacity
+		return assocPath([ticketId], minimumCapacity, acc);
+	}, {});
+};
```
Comment (tn3rb) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils.ts:108]: beautiful job making that complex logic easy to read and understand 💯

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts (hunk 3 above)
Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts:69]: Already pushed the fix

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts (hunk 2 above)
Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts]: Already done

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts (hunk 4 above)
Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/data/useOnSubmitAssignments.ts]: Fixed

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils.ts (hunk 5 above)
Comment (nicolad) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils.ts:108]: agree, this is well-written code;
this speaks to me of high level of caring for reviewers and ease of future development/ refactoring

**Pull request merged**